components/page/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { TopBar } from '@components/top-bar'
+import { Footer } from '@components/footer'
+
+import { StyledPage } from './styled'
+
+interface PageProps {
+  isErrorPage?: boolean;
+  children?: React.ReactNode;
+  admin?: boolean;
+  wrap?: boolean;
+  innerPadding?: number | number[];
+  subNav?: React.ReactNode;
+  align?: string;
+  [prop: string]: any;
```

Comment:
I'm still figuring out this situation with TS & React - do we need to have this line? Can TS be smart about the fact that you're passing props to a child?

Reply:
It is indeed needed, as we're saying: this component accepts these properties... and is used as a dictionary (of other style props). TypeError if removed.

The UI library _could_ expose some types to extend (that include all the base styled system props), in order to type it explicitly. However, I think this raises more of a question about how we build components. Ideally, imo, we don't want to mix the styling of components too much with the data/functionality they accept, otherwise important props, `admin={true/false}`get lost amongst `borderTop="grey"`.

An exception to this may be style variants a component needs, but even think it's better to encapsulate with true `variant` props, rather than exposing the entire styled-system suite of props. Maybe this goes against the whole paradigm of styled-system though? But definitely think there's merit it decoupling styles to some degree.